### PR TITLE
Fix input argument ordering

### LINKED ISSUE #148: Input argument ordering reorders literal before complex
## Description

ComplexData inputs are ordered after literal inputs since their default is not None, but a ComplexData instance. 

## Environment

* Birdy version used, if any: 0.6.4

### PR DESCRIPTION
**Pull request by huard**
## Overview

This PR fixes #148 (I think...)

Cannot really test with Emu as the recent authentication changes have broken the test suite for me.

### CODE REVIEW

File: tests/test_client.py [hunk 1]
```diff
@@ -255,7 +265,14 @@ def test_xarray_converter(wps):
     assert isinstance(ncdata, xr.Dataset)
 
 
-def test_sort_inputs():
+def test_sort_inputs(process):
+    # The first three inputs are all minOccurs=1 with no default, so we expect them to remain in the same order.
+    ps = sorted(process.dataInputs, key=sort_inputs_key)
+    for i in range(3):
+        assert ps[i] == process.dataInputs[i]
```
Comment (davidcaron) [inline tests/test_client.py]: I think this willl not work in python 3. Maybe this:
```python
    ps = sorted(process.dataInputs, key=sort_inputs_key)
    for i1, i2 in zip(ps, process.dataInputs):
        assert i1 == i2
```

**Comment by davidcaron** [comment]
I didn't test it but I think it makes sense how you did it... I'll try to test using the emu test suite later today, and try to see if owslib 0.18 works on my side for the `auth` argument issue.

File: tests/test_client.py (hunk 1 above)
Comment (huard) [reply tests/test_client.py]: Seems to be working with 3.6.3 at the very least.

Comment (davidcaron) [reply tests/test_client.py]: wow sorry, I was sure `sorted` returned a generator in python 3...

**Comment by cehbrecht** [comment]
@huard Is the `defaultValue` of a `ComplexInput` always a `ComplexData`? Would be the check then for the input type (literal or complex)?

**Comment by huard** [comment]
That's my understanding, yes. 
What do you mean ?

File: birdy/client/base.py [hunk 2]
```diff
@@ -344,9 +344,13 @@ def sort_inputs_key(i):
...
     i: owslib.wps.Input
       An owslib Input
+
+    Notes
+    -----
+    The defaultValue for ComplexData is ComplexData instance specifying mimetype, encoding and schema.
     """
     conditions = [
-        i.minOccurs >= 1 and i.defaultValue is None,
+        i.minOccurs >= 1 and (i.defaultValue is None or isinstance(i.defaultValue, ComplexData)),
```
Comment (cehbrecht) [inline birdy/client/base.py]: `... or isinstance(i, ComplexDataInput)`

see here:
https://github.com/geopython/OWSLib/blob/f48d8f69ada2a746072b16c8b3cba1b62c5fd1ec/owslib/wps.py#L178

Comment (cehbrecht) [reply birdy/client/base.py]: in case of ComplexInput checking the type of the input ... not the defaultValue type.

Comment (huard) [reply birdy/client/base.py]: Hum, this doesn't seem to work. The input is an `wps.Input` object, and the default value a `wps.ComplexData` object. So `is_complexdata` does not work on either.

Comment (cehbrecht) [reply birdy/client/base.py]: Yes, you are right. The names (`Input`, `ComplexDataInput`) are just confusing :)

**Pull request merged**